File: wastelessapp/src/app/models/grocery-list.ts
export class GroceryList {
  public GroceryListId: number;
  public GroceryListName: string;

  constructor(glName:string, glId?:number) {
    this.GroceryListName = glName;
    if (glId){
      this.GroceryListId = glId;
    }
  }

}

File: WasteLESS/WasteLESS/Models/GroceryList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations.Schema;

namespace WasteLESS.Models
{
    public class GroceryList
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int GroceryListId { get; set; }
        public string GroceryListName { get; set; }
        public IList<GroceryListItem> GroceryListItems { get; set; }
    }
}
